src/Components/Search.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+const searchSchema = z.object({
+  search: z.string(),
+});
+
+type SearchFormData = z.infer<typeof searchSchema>;
+
+function Search() {
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<SearchFormData>({
+    resolver: zodResolver(searchSchema),
+  });
+
+  const onSubmit = (data: SearchFormData) => {
+
+    console.log('Search Term:', data.search);
+    reset({ search: '' }); 
+  };
+  
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <label className='mr-2' >Search:             </label>
+        <Controller
+          name="search" 
+          control={control}
+          render={({ field }: { field: any }) => (
```

Comment:
```suggestion
          render={({ field }) => (
```
Never use any ever. Also you don't need to give it a type yourself.  It automatically has inferred the type. Not sure why you tried forcing to give it a type yourself.